assets/model_monitoring/components/src/shared_utilities/df_utils.py

```diff
@@ -34,6 +38,98 @@ def get_categorical_columns(column_dtype_map: dict) -> list:
     return categorical_columns
 
 
+def is_numerical_new(column, column_dtype_map: dict, feature_type_override_map: dict, df):
+    """Check if int column should be numerical."""
+    if feature_type_override_map.get(column, None) == "numerical":
+        return True
+    if feature_type_override_map.get(column, None) == "categorical":
+        return False
+    if column_dtype_map[column] in data_type_numerical_group:
+        return True
+    if column_dtype_map[column] in data_type_categorical_group:
+        return False
+    if column_dtype_map[column] in data_type_long_group:
+        distinct_value_ratio = get_distinct_ratio(df.select(column).rdd.flatMap(lambda x: x).collect())
+        return distinct_value_ratio >= 0.05
+
+    print(f"Unknown column type: {column_dtype_map[column]}, column name: {column}")
+    return False
+
+
+def is_categorical_new(column, column_dtype_map: dict, feature_type_override_map: dict, df):
+    """Check if int column should be categorical."""
+    if feature_type_override_map.get(column, None) == "categorical":
+        return True
+    if feature_type_override_map.get(column, None) == "numerical":
+        return False
+    if column_dtype_map[column] in data_type_categorical_group:
+        return True
+    if column_dtype_map[column] in data_type_numerical_group:
+        return False
+    if column_dtype_map[column] in data_type_long_group:
+        distinct_value_ratio = get_distinct_ratio(df.select(column).rdd.flatMap(lambda x: x).collect())
+        return distinct_value_ratio < 0.05
+
+    print(f"Unknown column type: {column_dtype_map[column]}, column name: {column}")
```

Comment:
you may want to log this, so you could have telemetry on which datatypes you're missing